apollo-core/src/main/java/com/ctrip/framework/foundation/internals/NetworkInterfaceManager.java

```diff
@@ -98,7 +99,19 @@ private void load() {
 
     try {
       Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
-      List<NetworkInterface> nis = interfaces == null ? Collections.<NetworkInterface>emptyList() : Collections.list(interfaces);
+      List<NetworkInterface> nis = interfaces == null ? Collections.<NetworkInterface>emptyList()
+          : Collections.list(interfaces);
+      //sort the network interfaces according to the index asc
+      Collections.sort(nis, new Comparator<NetworkInterface>() {
+        @Override
+        public int compare(NetworkInterface nis1, NetworkInterface nis2) {
+          if (nis1.getIndex() > nis2.getIndex()) {
+            return 1;
+          } else {
+            return -1;
```

Comment:
do we need a case to return 0?

Reply:
Return 0 is invalid。

this is results on my computer。
return 0:
name:utun0 (utun0)------index = 18
name:awdl0 (awdl0)------index = 12
name:en5 (en5)------index = 8
name:en0 (en0)------index = 10
name:lo0 (lo0)------index = 1

return -1:
name:lo0 (lo0)------index = 1
name:en5 (en5)------index = 8
name:en0 (en0)------index = 10
name:awdl0 (awdl0)------index = 12
name:utun0 (utun0)------index = 18

I don't know why return 0 sorting does not take effect.....

Reply:
normally we would return 3 values for a comparison between variable `a` and `b`
1. `1` means a > b
2. `0` means a = b
3. `-1` means a < b

I'm not saying we should change the statement `return -1` to `return 0`, what I mean by `do we need a case to return 0?` is I'm not sure if there is a case that nis1.getIndex() == nis2.getIndex()

Reply:
I know.Maybe this can be replaced by "Integer.compare(nis1.getIndex(), nis2.getIndex())"